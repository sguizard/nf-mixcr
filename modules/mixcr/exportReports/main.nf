process MIXCR_EXPORTREPORTS {
    tag "$meta.id"
    container 'ghcr.io/milaboratory/mixcr/mixcr:4.6.0'

    input:
    tuple val(meta), path(clns)
    path(library)

    output:
    path("*.report.json"), emit: exportReports_json
    path("*.report.txt") , emit: exportReports_txt
    path "versions.yml"  , emit: versions

    script:
    def args   = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: meta.id

    """
    mixcr exportReports --json $clns ${prefix}.report.json
    mixcr exportReports $clns ${prefix}.report.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        MiXCR: \$( mixcr --version | grep -e 'MiXCR' | sed 's/MiXCR v//' | sed 's/ .\\+//' )
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: id
    """
    touch ${prefix}.report.json
    touch ${prefix}.report.txt

    cat <<-END_VERSIONS > versions.yml
    MIXCR_ANALYZE:
        MiXCR: \$( mixcr --version | grep -e 'MiXCR' | sed 's/MiXCR v//' | sed 's/ .\\+//' )
    END_VERSIONS
    """
}
