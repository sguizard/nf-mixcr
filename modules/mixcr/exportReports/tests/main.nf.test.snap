{
    "MIXCR_EXPORTREPORTS Compute and export reports": {
        "content": [
            {
                "0": [
                    "[id:TCR_cat].report.json:md5,f8c54b166605a984bb9902ed7380f928"
                ],
                "1": [
                    "[id:TCR_cat].report.txt:md5,01e156da28e7d11b821e520e8dc32359"
                ],
                "2": [
                    "versions.yml:md5,73d3c14d87591f484cdc03d1a32ccc17"
                ],
                "exportReports_json": [
                    "[id:TCR_cat].report.json:md5,f8c54b166605a984bb9902ed7380f928"
                ],
                "exportReports_txt": [
                    "[id:TCR_cat].report.txt:md5,01e156da28e7d11b821e520e8dc32359"
                ],
                "versions": [
                    "versions.yml:md5,73d3c14d87591f484cdc03d1a32ccc17"
                ]
            }
        ],
        "timestamp": "2024-01-22T14:51:26.154291"
    }
}